api/v1alpha1/ipv4lease_types.go

```diff
@@ -19,75 +19,66 @@ package v1alpha1
 import (
 	"time"
 
-	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-// IPLeaseSpec defines the desired state of IPLease
-type IPLeaseSpec struct {
-	// References the pool to lease an IP from.
+// IPv4LeaseSpec defines the desired state of IPv4Lease
+type IPv4LeaseSpec struct {
+	// References the IPv4Pool to lease an IP from.
 	Pool LocalObjectReference `json:"pool"`
+
+	// Type of the IPv4Lease.
+	// +kubebuilder:default="Dynamic"
+	// +kubebuilder:validation:Enum=Dynamic;Static
+	Type IPLeaseType `json:"type"`
+
 	// Static IP lease settings.
-	Static *StaticIPLease `json:"static,omitempty"`
-	// IPFamilies that this lease should include.
-	IPFamilies []corev1.IPFamily `json:"ipFamilies"`
+	Static *IPLeaseStatic `json:"static,omitempty"`
+
 	// Renew time is the time when the lease holder has last updated the lease.
 	// Falls back to .metadata.creationTimestamp if not set.
```

Comment:
This is not true for kubectl:

![image](https://user-images.githubusercontent.com/5539202/110188420-ebc97f80-7e1b-11eb-8ccb-6f24bd1ceba6.png)

Reply:
I think `.spec.renewTime` is unused